### PR TITLE
Add Package Control support

### PR DESCRIPTION
**Pull request by Lykegenes**
Hi!

I added PackageControl to the supported providers. I copied the code from the Packagist block right above since their behaviour is the same.

Here is an example of the response format : https://packagecontrol.io/packages/Laravel%205%20Snippets.json

Also, I haven't been able to test it, since I don't have a development environment setup.

Thanks!

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1210,6 +1210,37 @@ cache(function(data, match, sendBadge, request) {
...
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+    }
+    try {
+      var data = JSON.parse(buffer);
+
+      var downloads = data.package.installs.total;
+      badgeData.text[1] = metric(downloads) + ' total';
+
```
Comment (espadrine) [inline server.js]: Please remove spaces at the end of lines.

File: server.js [hunk 2]
```diff
@@ -1210,6 +1210,37 @@ cache(function(data, match, sendBadge, request) {
...
+  }
+  request(apiUrl, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+    }
+    try {
+      var data = JSON.parse(buffer);
+
+      var downloads = data.package.installs.total;
```
Comment (espadrine) [inline server.js]: I believe it should be `data.installs.total` instead of `data.package.installs.total`.

File: server.js [hunk 3]
```diff
@@ -1210,6 +1210,37 @@ cache(function(data, match, sendBadge, request) {
...
+      badgeData.text[1] = metric(downloads) + ' total';
```
Comment (espadrine) [inline server.js]: Could you omit `total`? We'll consider that this is the default assumption.

File: server.js [hunk 4]
```diff
@@ -1210,6 +1210,37 @@ cache(function(data, match, sendBadge, request) {
...
 }));
 
+// Package Control integration.
+camp.route(/^\/packagecontrol\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var userRepo = match[1];  // eg, `Package%20Control`.
+  var format = match[2];
+  var apiUrl = 'https://packagecontrol.io/packages/' + userRepo + '.json';
+  var badgeData = getBadgeData('downloads', data);
+  if (userRepo.substr(-14) === '/:package_name') {
```
Comment (espadrine) [inline server.js]: What is the purpose of this test?

File: server.js [hunk 5]
```diff
@@ -1210,6 +1210,37 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Package Control integration.
+camp.route(/^\/packagecontrol\/(.*)\.(svg|png|gif|jpg|json)$/,
```
Comment (espadrine) [inline server.js]: Could you add a prefix to distinguish between different packagecontrol badges? Maybe `/packagecontrol/dt` for total downloads (we might have `/packagecontrol/dd` for daily downloads, etc.)?

File: server.js [hunk 6]
```diff
@@ -1210,6 +1210,37 @@ cache(function(data, match, sendBadge, request) {
...
+cache(function(data, match, sendBadge, request) {
+  var userRepo = match[1];  // eg, `Package%20Control`.
```
Comment (espadrine) [inline server.js]: Could you add the corresponding badge (`https://img.shields.io/packagecontrol/dt/Package%20Control.svg`) to `try.html`, please?

**Comment by espadrine** [comment]
Thanks a lot for this contribution!

File: server.js (hunk 4 above)
Comment (Lykegenes) [inline server.js]: I copied it from line 1174 just above. I don't think it does anything so I'll remove it.

**Comment by Lykegenes** [comment]
It's fixed, and I added daily/weekly/monthly downloads

**Comment by Lykegenes** [comment]
Done! Will it be added automatically or not at all? Other services have it in their badges

**Comment by espadrine** [comment]
That's fantastic!

`total` will be removed from other badges. I feel it was a mistake of me to have accepted those with `total` in the first place.

**Comment by espadrine** [comment]
As usual, the badge will be live, but the website will only show the new badge after 24h, in case we realize we did a really bad mistake.

**Comment by Lykegenes** [comment]
Great, thanks! Will keep that in mind in the future.

**Pull request merged**